File: 프로그래머스/lv2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    
    for i in range(yellow + 1):
        if i == 0:
            continue
                    
        if yellow % i != 0:
            continue

        
        if (yellow / i + 2) * (i + 2)  == brown + yellow:
            return [yellow / i + 2, (i + 2)]
    
    return answer

File: 프로그래머스/lv2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        
        for (int h = 1; h <= yellow; h++) {
            double w = (double)yellow / (double)h;
            if (!isInteger(w)) {
                continue;
            }
            int intW = (int)w;
            if((h + 2) * (intW + 2) == brown + yellow) {
                answer[0] = intW + 2;
                answer[1] = h + 2;
                break;
            }
        }
        return answer;
    }
    
    private boolean isInteger(double n) {
        return n == (int)n;
    }
}